addons/modules/functions/fnc_moduleTracers.sqf

```diff
@@ -0,0 +1,129 @@
+deleteVehicle _gunner;
+
+_gunner = createAgent ["B_Soldier_F", [0, 0, 0], [], 0, "NONE"];
+_gunner attachTo [_logic, [0, 0, 0]];
+_gunner hideObjectGlobal true;
+_gunner allowDamage false;
+_gunner setCaptive true;
+_gunner switchMove "AmovPercMstpSrasWrflDnon";
+_gunner disableAI "ANIM";
+_gunner disableAI "MOVE";
+_gunner disableAI "TARGET";
+_gunner disableAI "AUTOTARGET";
+_gunner setBehaviour "CARELESS";
+_gunner setCombatMode "BLUE";
+
+_gunner setUnitLoadout (configFile >> "EmptyLoadout");
+_gunner addWeapon _weapon;
+_gunner selectWeapon _weapon;
+_gunner addWeaponItem [_weapon, _magazine, true];
+
+_logic setVariable [QGVAR(tracersGunner), _gunner];
+
+[{
+    params ["_args", "_pfhID"];
+    _args params ["_logic", "_gunner", "_target", "_weapon", "_delay", "_dispersion", "_nextBurstTime"];
+
+    if (!alive _logic || {!alive _gunner}) exitWith {
+        [_pfhID] call CBA_fnc_removePerFrameHandler;
+        deleteVehicle _gunner;
+    };
+
+    if (
+        CBA_MissionTime >= _nextBurstTime && {
+            (playableunits + switchableunits) findIf {_gunner distance _x < 100} == -1
+        }
+    ) then {
+        if (_target isEqualType "") then {
+            _target = call compile _target;
+        };
+        if (isNil "_target") then {_target = objNull};
+
+        private _vectorToTarget = [0, 0, 0];
+        private _targetPos = [0, 0, 0];
+        private _logicPos = getPosASLVisual _logic;
+        private _dir = 0;
+        private _pitch = 0;
+
+        // Sets vector to the target if it's specified
+        if (!(_target isEqualTo objNull)) then {
+            // Refresh target
+            if (_target isEqualType objNull) then {
+                _targetPos = getPosASLVisual _target;
+            } else {
+                _targetPos = _target;
+            };
+            _vectorToTarget = _logicPos vectorFromTo _targetPos;
+
+            // Vector randomization
+            _vectorToTarget = _vectorToTarget vectorAdd [random [-_dispersion, 0, _dispersion], random [-_dispersion, 0, _dispersion], random [-_dispersion, 0, _dispersion]];
+            _logic setVectorDirAndUp [_vectorToTarget, _vectorToTarget vectorCrossProduct [-(_vectorToTarget # 1), _vectorToTarget # 0, 0]];
+        } else {
+            // Random firing (old behavior)
+            _dir = -5 + random 10;
+            _pitch = 30 + random 60;
+            _gunner setdir (random 360);
+            [_gunner, _pitch, 0] call BIS_fnc_setpitchbank;
+        };
+
+        private _shotDelay = 0.05 + random 0.1;
+        private _burstLength = 0.1 + random 0.9;
+
+        [{
+            params ["_nextShotTime", "_logic", "_gunner", "_dispersion", "_weapon", "_shotDelay"];
+
+            // Aim
+            if (!(_target isEqualTo objNull)) then {
+                _vectorToTarget = _vectorToTarget vectorAdd [random [-_dispersion, 0, _dispersion], random [-_dispersion, 0, _dispersion], random [-_dispersion, 0, _dispersion]];
```

Comment:
This variable is undefined in this `CBA_fnc_waitUntilAndExecute` code block.

Reply:
Thank you! I think I've done that and cleaned up my old code a little.